src/flask/app.py

```diff
@@ -435,8 +436,7 @@ def __init__(
         #: you how often they got attached.
         #:
         #: .. versionadded:: 0.7
-        self.blueprints = {}
-        self._blueprint_order = []
+        self.blueprints = OrderedDict()
```

Comment:
I think we can assume dicts are ordered for all Pythons we support. Dicts are ordered in PyPy 3.6 and CPython 3.6 as part of the implementation, and this became part of the language in 3.7. Neither Jython nor IronPython support Python 3.

Reply:
The commit message says you're keeping `OrderedDict` to indicate that order matters, but given that it's a fairly internal API and we're not taking advantage of any of the differences enumerated in the docs, I don't think it's worth it right now. We can call out in the docstring that the registration order is retained.